### DISCUSSION TITLE
Text replacement in pdf

### DISCUSSION THREAD

**Discussion by presvetli** [pymupdf/PyMuPDF#2512] [Looking for help]
Is there some simple way to replace some word/sentence in PDF and save it? Of course, I want keep old format of PDF with all tables and images.

**Comment by JorjMcKie** [comment]
The default approach is using redaction annotations.
It erases text with a given rectangle and optionally also punches holes in overlapping images.

Plus you may specify text that should be written in that area. This latter feature yields approximate solution only, because of potentially imprecise new text positioning and a limitation of usable fonts.
But you can use a two 2-phase approach:
1. Extract the text you want to erase using the "dict" option. This provides you all properties: font, font size, color, etc.
2. Erase the text via redaction procedure.
3. Insert the new text using applicable properties of the original text.

**Reply by presvetli** [reply]
thank you.

**Reply by JorjMcKie** [reply]
Please do not hesitate to ask if you run into problems.
